guides/dsl.md

```diff
@@ -0,0 +1,314 @@
+# Eve JavaScript/TypeScript DSL
+
+The forthcoming v0.3 release of Eve supports a domain specific langauge (DSL) that interacts with the Eve server, allowing you to intermix Javascript and Eve code. There are several benefits to using the DSL:
+
+1) The syntax is native Javascript, so if you know JS you can write in the DSL
+2) You can interact with Javascript functions and libraries
+3) The DSL can be used "a la carte", so you can use as much of Eve as you need to for your project. Thus you can write an entire application in Eve, just use it as a datastore, or anywhere between. This makes it easy to integrate Eve with an existing Javascript application.
+4) You can easily import data into Eve. If you can get your data in Javascript, you can use it in Eve.
+
+This DSL guide is for users who are already familiar with Eve semantics. For those new to Eve, we'll have more fundamental guides released in the near future.
+
+## Summary of DSL 
+
+- find a record: `find("person", {salary})` <-> `[#person salary]`
+- bind/commit a record: `record("person", {salary})` <-> `[#person salary]`
+- not: `not()` <-> `not()`
+- choose: `choose()` <-> `if-then / else if-then`
+- union: `union()` <-> `if-then / if-then`
+- Add a value: `person.add("salary", 10)` <-> `person.salary += 10`
+- Remove a value: `person.remove("salary, 10)` <-> `person.salary -= 10`
+- Set a value: `person.remove("salary").add("salary", 10)` <-> `person.salary := 10`
+- Remove an attribute `person.remove("salary")` <-> `person.salary := none`
+- Remove a record `person.remove()` <-> `person := none`
+- functions `lib.math.sin()` <-> `sin[]`
+- aggregates `gather().pers().count()` <-> `count[given per]` 
+
+## Using the DSL
+
+
+The Eve Javascript DSL is requires ES6. For added readability, we make frequent use of [destructuring][1] and [arrow functions][2] available in TypeScript 2.1.0 or later.
```

Comment:
It doesn't require ES6, you can write it normally in ES5, we've just chosen to write in ES6 and thus the guide is written with ES6.